### ISSUE TITLE
Spurious warnings generated

### ISSUE THREAD

**Issue by edwardalee** [lf-lang/lingua-franca#1706]
Compiling anything with the C target results in a barrage of warnings like this:
```
WARNING: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: libcore.a(scheduler_adaptive.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: libcore.a(scheduler_GEDF_NP_CI.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: libcore.a(scheduler_GEDF_NP.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: libcore.a(scheduler_PEDF_NP.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: libcore.a(modes.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: libcore.a(lf_linux_support.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: libcore.a(lf_windows_support.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: libcore.a(lf_nrf52_support.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: libcore.a(lf_zephyr_support.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: libcore.a(scheduler_adaptive.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: libcore.a(scheduler_GEDF_NP_CI.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: libcore.a(scheduler_GEDF_NP.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: libcore.a(scheduler_PEDF_NP.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: libcore.a(modes.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: libcore.a(lf_linux_support.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: libcore.a(lf_windows_support.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: libcore.a(lf_nrf52_support.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: libcore.a(lf_zephyr_support.c.o) has no symbols
WARNING: /Users/eal/git/examples-lingua-franca/C/src/browser-ui/BrowserUI.lf line 1
Warning in imported file: /Users/eal/git/examples-lingua-franca/C/src/browser-ui/BrowserUI.lf
```

This has been around for a while, but it really looks bad. This needs to be fixed before we can release again.

**Comment by petervdonovan** [comment]
In case anyone forgot, this happens because we use the preprocessor to decide whether to `#include` code. So, we know that the files "having no symbols" is the intended behavior, but we just need to suppress the warning.

**Comment by lhstrh** [comment]
Hasn't this been fixed?

**Comment by petervdonovan** [comment]
Yes. Apparently this is also a duplicate of https://github.com/lf-lang/reactor-c/issues/132.
